### DISCUSSION TITLE
Splendid !

### DISCUSSION THREAD

**Discussion by Gluups** [community/community#22645] [New to GitHub]
<p>Magnificent ! After a few days or development I decided to create a repository to be very sure not to lose anything. As soon as I clicked on Push, everything has instantly disappeared. I presume this is not the aim of Git. Is there an operational mode somewhere, other than the one that led me there ?</p>

**Comment by Gluups** [comment]
<p>I think I should reformulate my question (I did not see any “Edit” button, is there ?) Did anyone already succeed in pushing an existing project into a Git repository, that was created afterwards ?</p>

**Comment by Gluups** [comment]
<p>So I realized I had to create a new project, and link it to my repository before doing any modification in it. Maybe I shall do the development in another project, and transfer it from time to time to the linked project, to avoid side effects.</p>
<p>Well but when linking the project to the repository just after creating the project, I get an error :</p>
<blockquote>
<p>“rejected Updates were rejected because the remote contains work that you do not have locally. This is usually caused by another repository pushing to the same ref. You may want to first integrate the remote changes before pushing again.”</p>
</blockquote>
<p>Well, my repository contains .gitattributes, and .gitignore, and that is all. And they are empty.</p>
<p>Is the solution to suppress the repository and create a new one ?</p>
<p>If there are incoming changes, why not propose me to pull them ?</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>This latest error is basically saying that you:</p>
<ol>
<li>Created a Git repository with a history A-B-C (on GitHub)</li>
<li>Created a Git repository with a history X-Y-Z (locally)</li>
<li>Then told Git to push history X-Y-Z onto A-B-C</li>
</ol>
<p>Because there’s no common point of reference, Git can’t figure out in an automated fashion what it is likely you want. So the error is suggesting that you pull the A-B-C history down, manually sort out what you want, and then push the result back up.</p>
<p>Or the faster way would possibly be to delete the repository on GitHub, create a new one with <strong>no</strong> contents, then push your local contents up to the new repository.</p>
<p>I hope that helps!</p>

**Comment by DianaWills** [comment]
Thank you for sharing this aspect of creating a repository. I was also planning to create one for my project [LED video wall rental](https://www.unilightled.com/rental-led/). I should better discuss it with some professional before taking this step.

**Comment by assistalandlord** [comment]
Ouch, that sounds rough! Git definitely isn’t meant to evict your work like that—unless something went wrong with your push or branch settings. It’s a bit like calling a [commercial eviction service](https://www.assistalandlord.co.uk/commercial-eviction-and-crar/) when all you needed was a quick fix. Hopefully, you can recover your files with some Git sleuthing!
